cardano-cli/src/Cardano/CLI/Run/Mnemonic.hs

```diff
@@ -0,0 +1,184 @@
+      Cmd.ExtendedSigningCCColdKey ->
+        writeKeyToFile
+          =<< wrapException
+            (signingKeyFromMnemonic AsCommitteeColdExtendedKey mnemonicWords derivationAccountNo)
+      Cmd.ExtendedSigningCCHotKey ->
+        writeKeyToFile
+          =<< wrapException
+            (signingKeyFromMnemonic AsCommitteeHotExtendedKey mnemonicWords derivationAccountNo)
+ where
+  writeSigningKeyFile
+    :: (HasTextEnvelope (SigningKey a), SerialiseAsBech32 (SigningKey a))
+    => KeyOutputFormat -> SigningKeyFile Out -> SigningKey a -> ExceptT KeyCmdError IO ()
+  writeSigningKeyFile fmt sKeyPath skey =
+    firstExceptT KeyCmdWriteFileError $
+      case fmt of
+        KeyOutputFormatTextEnvelope ->
+          newExceptT $
+            writeLazyByteStringFile sKeyPath $
+              textEnvelopeToJSON Nothing skey
+        KeyOutputFormatBech32 ->
+          newExceptT $
+            writeTextFile sKeyPath $
+              serialiseToBech32 skey
+
+  readMnemonic :: Cmd.MnemonicSource -> ExceptT KeyCmdError IO [Text]
+  readMnemonic (Cmd.MnemonicFromFile filePath) = do
+    fileText <- firstExceptT KeyCmdReadMnemonicFileError $ except =<< readTextFile filePath
+    return $ map T.pack $ words $ T.unpack fileText
+  readMnemonic Cmd.MnemonicFromInteractivePrompt =
+    liftIO $ do
+      putStrLn $
+        unlines
+          [ ""
+          , "Please enter your mnemonic sentence."
+          , ""
+          , " - It should consist of either: 12, 15, 18, 21, or 24 words."
+          , " - To terminate, press enter on an empty line."
+          , " - To abort you can press CTRL+C."
+          , ""
+          , "(If your terminal supports it, you can use the TAB key for word completion.)"
+          , ""
+          ]
+      runInputTBehaviorWithPrefs defaultBehavior defaultPrefs settings (inputT ("", "") [])
+   where
+    settings :: Monad m => Settings m
+    settings =
+      Settings
+        { complete = completionFunc
+        , historyFile = Nothing
+        , autoAddHistory = False
+        }
+
+    completionFunc :: Monad m => CompletionFunc m
+    completionFunc = completeWord' Nothing isSpace completeMnemonicWord
+
+    completeMnemonicWord :: Monad m => String -> m [Completion]
+    completeMnemonicWord prefix = return $ map (simpleCompletion . T.unpack . fst) $ findMnemonicWordsWithPrefix (T.pack prefix)
+
+    inputT :: (String, String) -> [Text] -> InputT IO [Text]
+    inputT prefill mnemonic = do
+      minput <- getInputLineWithInitial (show (length mnemonic + 1) <> ". ") prefill
+      case minput of
+        Nothing -> return $ reverse mnemonic
+        Just "" -> return $ reverse mnemonic
+        Just input ->
+          let newWords = map (T.toLower . T.pack) $ filter (not . null) $ words input
+           in case span isValidMnemonicWord newWords of
+                (allWords, []) -> inputT ("", "") (reverse allWords ++ mnemonic)
+                (validWords, invalidWord : notValidatedWords) -> do
```

Comment:
should `notValidatedWords` be `remainingInvalidWords`?

Reply:
No, they may all be valid, we just haven't checked. `span` stops at the first `False`

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/pull/975/commits/eab669bc012c77743285fddabd3492933646866a